static/js/portico/recaptcha.js

```diff
@@ -0,0 +1,36 @@
+var recaptcha = (function () {
+
+var exports = {};
+
+var form_id;
+
+exports.init = function (fid) {
+    form_id = fid;
+    $("#" + form_id).find(":submit").click(function () {
+        if (grecaptcha === undefined) {
+          // If reCAPTCHA library was not loaded, stop processing and return.
+          return true;
+        }
+
+        var form = document.getElementById(form_id);
+        if (form.checkValidity()) {  // Do client side validation.
+            grecaptcha.execute();
+            // Don't submit form from here. reCAPTCHA will call onSubmit after
+            // validation.
+            return false;
+        }
+    });
+    $.getScript("https://www.google.com/recaptcha/api.js");
+};
+
+exports.on_submit = function (response) {
+    $('#g-recaptcha-response').text(response);  // Add the response to form.
```

Comment:
What's this for? I don't see it in the examples in the reCAPTCHA docs.

Reply:
So after validation reCAPTCHA adds a field called 'g-recaptcha-response' in the form. It is filled if we use the normal flow, but I guess since we are using programmatic flow, it is not filled in. This line adds the response token to that field.

Reply:
This isn't in the examples in the docs, including for the flow we're using. Also when I tried removing this line and reloading, the interaction worked just fine.

What did you observe that caused you to add this line?

Reply:
I was getting "Missing token error", so I debugged and found out that the field wasn't being populated. This is why I added this. Maybe I missed something. Let me give it another iteration.

Reply:
OK, it did work without the line. Just checked it. Updated.